pkg/framework/plugins/removepodsviolatinginterpodantiaffinity/pod_antiaffinity.go

```diff
@@ -78,6 +78,13 @@ func (d *RemovePodsViolatingInterPodAntiAffinity) Name() string {
 }
 
 func (d *RemovePodsViolatingInterPodAntiAffinity) Deschedule(ctx context.Context, nodes []*v1.Node) *framework.Status {
+	allPods, err := d.listExistingPods(ctx)
+	if err != nil {
+		return &framework.Status{
+			Err: fmt.Errorf("error listing all pods: %v", err),
+		}
+	}
+	nodeMap := createNodeMap(nodes)
```

Comment:
could `nodeMap` replace the need to call `ListPodsOnANode()` in the loop below?

Reply:
or, maybe not nodeMap sorry... but looking at how we call `allPods` above and then call `ListPodsOnANode`, maybe we can cut out one of them

Reply:
Thank you for your review and sorry for my late reply.

I removed `ListPodsOnANode` call and keep code to query for all pods. Because listing pods by each node looks not efficient.
But, I'm not sure that my code filtering pods for `podsOnANode` does same as `ListPodsOnANode` call.